src/cljs/rems/handlers.cljs

```diff
@@ -36,3 +37,18 @@
  :set-current-language
  (fn [db [_ language]]
    (assoc db :language language)))
+
+(reg-event-fx
+ :landing-page-redirect!
+ (fn [{:keys [db]}]
+   (if (get-in db [:identity :roles])
+     (let [roles (set (get-in db [:identity :roles]))]
+       (println "Selecting landing page based on roles" roles)
+       (.removeItem js/window.sessionStorage "rems-redirect-url")
+       (cond
+         (contains? roles :owner) (dispatch! "/#/administration")
+         (contains? roles :approver) (dispatch! "/#/actions")
+         (contains? roles :reviewer) (dispatch! "/#/actions")
+         :else (dispatch! "/#/catalogue"))
+       {})
+     {:dispatch [:landing-page-redirect!]})))
```

Comment:
are you rerunning the same event here?

Reply:
This event requires set-identity to have been processed first and since events may be processed in random order (esp. Firefox), so we'll dispatch the same event for the next "round" if there is no roles set yet.

Reply:
maybe a comment to this effect would be good